app/src/main/java/org/eyeseetea/malariacare/SplashScreenActivity.java

```diff
@@ -64,20 +60,38 @@ private void init() {
             Session.setMaxTotalQuestions(Program.getMaxTotalQuestions());
         }
 
-        try {
-            if (!BuildConfig.multiuser) {
-                Log.i(TAG, "Creating demo login from dashboard ...");
-                LoginUseCase loginUseCase = new LoginUseCase(this);
+        if (!BuildConfig.multiuser) {
+            Log.i(TAG, "Creating demo login from dashboard ...");
+            IUserAccountRepository userAccountRepository = new UserAccountRepository(this);
+            LoginUseCase loginUseCase = new LoginUseCase(userAccountRepository);
+
+            Credentials demoCrededentials = Credentials.createDemoCredentials();
 
-                Credentials demoCrededentials = Credentials.createDemoCredentials();
+            loginUseCase.execute(demoCrededentials, new ALoginUseCase.Callback() {
+                @Override
+                public void onLoginSuccess() {
+                    Log.d(TAG, "Login Success");
 
-                loginUseCase.execute(demoCrededentials);
-            }
+                    //TODO: execute PullUseCase for laos and cambodia here
```

Comment:
@idelcano  yes, laos and cambodia code in this moment is not being refactored and tested, this TODO is for future